nextcord/embeds.py

```diff
@@ -427,17 +456,79 @@ def set_image(self: E, url: MaybeEmpty[Any]) -> E:
         -----------
         url: :class:`str`
             The source URL for the image. Only HTTP(S) is supported.
+        file: Optional[File]
+            A file to use for the image.
+
+            .. versionadded:: 2.0
         """
+        from . import InvalidArgument
 
         if url is EmptyEmbed:
             try:
                 del self._image
             except AttributeError:
                 pass
+
+        if file is EmptyEmbed:
+            try:
+                del self._local_files["image"]
```

Comment:
this might be worth it to use [`dict.pop(item, default)`](https://docs.python.org/3/library/stdtypes.html#dict.pop) where `default` is `None` so a `ValueError` is not raised